controllers/openshift_controller_test.go

```diff
@@ -609,4 +609,31 @@ var _ = Describe("OpenShift KataConfig Controller", func() {
 
 		})
 	})
+	Context("Custom KataConfig with CheckNodeReadiness create", func() {
```

Comment:
Probably need to have a note / issue somewhere to track that you are not yet testing the NFD scenario.

Reply:
The commit mentions it's a basic checkNodeReadiness test. Adding additional tests is doable but I think its better to run it against real cluster. Nonetheless I'll work on adding more unit tests as separate PRs. Hope it's fine.